framework/src/base/MooseApp.C

```diff
@@ -88,240 +89,250 @@ void
+      "use_executor", "--executor", "Use the new Executor system instead of Executioners");
 
   params.addCommandLineParam<bool>(
-      "apptype", "--type", false, "Return the name of the application object.");
+      "show_type", "--show-type", "Return the name of the application object");
+  params.addCommandLineParam<bool>("yaml", "--yaml", "Dumps all input file syntax in YAML format");
   params.addCommandLineParam<std::string>(
-      "yaml", "--yaml", "Dumps input file syntax in YAML format.");
+      "yaml_search", "--yaml-search", "Dumps input file syntax matching a search in YAML format");
+  params.addCommandLineParam<bool>("json", "--json", "Dumps all input file syntax in JSON format");
   params.addCommandLineParam<std::string>(
-      "json", "--json", "Dumps input file syntax in JSON format.");
+      "json_search", "--json-search", "Dumps input file syntax matching a search in JSON format");
   params.addCommandLineParam<bool>(
-      "syntax", "--syntax", false, "Dumps the associated Action syntax paths ONLY");
+      "syntax", "--syntax", "Dumps the associated Action syntax paths ONLY");
   params.addCommandLineParam<bool>(
-      "show_docs", "--docs", false, "print url/path to the documentation website");
+      "show_docs", "--docs", "Print url/path to the documentation website");
   params.addCommandLineParam<bool>("check_input",
                                    "--check-input",
-                                   false,
-                                   "Check the input file (i.e. requires -i <filename>) and quit.");
-  params.addCommandLineParam<std::string>(
+                                   "Check the input file (i.e. requires -i <filename>) and quit");
+  params.setGlobalCommandLineParam("check_input");
+  params.addCommandLineParam<bool>(
       "show_inputs",
       "--show-copyable-inputs",
-      "Shows the directories able to be installed (copied) into a user-writable location");
+      "Shows the directories able to be copied into a user-writable location");
 
-  params.addCommandLineParam<std::string>("copy_inputs",
-                                          "--copy-inputs <dir>",
-                                          "Copies installed inputs (e.g. tests, examples, etc.) to "
-                                          "an directory named  <appname>_<dir>.");
-  params.addCommandLineParam<std::string>("run",
-                                          "--run",
-                                          "Runs the inputs in the current directory copied to a "
-                                          "user-writable location by \"--copy-inputs\"");
+  params.addCommandLineParam<std::string>(
+      "copy_inputs",
+      "--copy-inputs <dir>",
+      "Copies installed inputs (e.g. tests, examples, etc.) to a directory <appname>_<dir>");
+  // TODO: Should this remain a bool? It can't be a regular argument because it contains
+  // values that have dashes in it, so it'll get treated as another arg
```

Comment:
I see you're special-casing that below. Would `std::string` not work here? I guess not. Maybe have a special type for it.

Reply:
Yeah, I changed this to a `std::string`. It's weird. Maybe a special type would be better.